lib/ecto/association.ex

```diff
@@ -432,9 +432,13 @@ defmodule Ecto.Association.Has do
     {key, value} = parent_key(assoc, parent)
     changeset = update_parent_key(changeset, action, key, value)
 
-    changeset = case parent.__meta__.source do
-      { nil, _ } -> changeset
-      { prefix, _ } -> %{changeset | data: Ecto.put_meta(changeset.data, prefix: prefix)}
+    changeset = if parent do
```

Comment:
This would be cleaner:

```elixir
changeset =
  case parent do
    %{__meta__: %{source: {prefix, _}}} -> update_in changeset.data, &Ecto.put_meta(&1, prefix: prefix)
    _ -> changeset
  end